lib/trie/trie.go

```diff
@@ -669,7 +683,12 @@ func (t *Trie) clearPrefix(cn node, prefix []byte) (node, bool) {
 		if len(prefix) == len(c.key)+1 && length == len(prefix)-1 {
 			// found prefix at child index, delete child
 			i := prefix[len(c.key)]
-			c.children[i] = nil
+
+			cpy := [16]node{}
+			copy(cpy[:], c.children[:])
```

Comment:
nit/FYI this is an array, so you can simply do `cpy := c.children`, no need to make and copy it :wink: Not the case for slices though.